grpclb.go

```diff
@@ -59,646 +69,266 @@ type balanceLoadClientStream struct {
 	ClientStream
 }
 
-func (x *balanceLoadClientStream) Send(m *lbmpb.LoadBalanceRequest) error {
+func (x *balanceLoadClientStream) Send(m *lbpb.LoadBalanceRequest) error {
 	return x.ClientStream.SendMsg(m)
 }
 
-func (x *balanceLoadClientStream) Recv() (*lbmpb.LoadBalanceResponse, error) {
-	m := new(lbmpb.LoadBalanceResponse)
+func (x *balanceLoadClientStream) Recv() (*lbpb.LoadBalanceResponse, error) {
+	m := new(lbpb.LoadBalanceResponse)
 	if err := x.ClientStream.RecvMsg(m); err != nil {
 		return nil, err
 	}
 	return m, nil
 }
 
-// NewGRPCLBBalancer creates a grpclb load balancer.
-func NewGRPCLBBalancer(r naming.Resolver) Balancer {
-	return &grpclbBalancer{
-		r: r,
-	}
+// NewLBBuilder creates a builder for grpclb.
+func NewLBBuilder() balancer.Builder {
+	// TODO(bar grpclb) this function is exported for testing only, remove it when resolver supports selecting grpclb.
+	return NewLBBuilderWithFallbackTimeout(defaultFallbackTimeout)
 }
 
-type remoteBalancerInfo struct {
-	addr string
-	// the server name used for authentication with the remote LB server.
-	name string
+// NewLBBuilderWithFallbackTimeout creates a grpclb builder with the given
+// fallbackTimeout.
+//
+// Only call this function when a non-default fallback timeout is needed.
```

Comment:
I assume the "testing only" applies here too?

Reply:
What's a fallback timeout?  OK, I think I know what it means, but can you explain in the comment?

Reply:
Done.